src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_EditProfileScreen.tsx

```diff
@@ -0,0 +1,296 @@
+import {ActivityIndicator, View} from 'react-native';
+import {Section, TextInputSectionItem} from '@atb/components/sections';
+import React, {useEffect, useState} from 'react';
+import {dictionary, useTranslation} from '@atb/translations';
+import {EditProfileTexts} from '@atb/translations/screens/subscreens/EditProfileScreen';
+import {FullScreenView} from '@atb/components/screen-view';
+import {ThemeText} from '@atb/components/text';
+import {StyleSheet} from '@atb/theme';
+import {useAuthState} from '@atb/auth';
+import {Button} from '@atb/components/button';
+import Delete from '@atb/assets/svg/mono-icons/actions/Delete';
+import {MessageBox} from '@atb/components/message-box';
+import parsePhoneNumber from 'libphonenumber-js';
+import {numberToAccessibilityString} from '@atb/utils/accessibility';
+import {isValidEmail} from '@atb/utils/validation';
+import {CustomerProfile} from '@atb/api/types/profile';
+import {useProfileQuery, useProfileUpdateMutation} from '@atb/queries';
+
+type EditProfileScreenProps = ProfileScreenProps<'Profile_EditProfileScreen'>;
+
+export const Profile_EditProfileScreen = ({
+  navigation,
+}: EditProfileScreenProps) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const {
+    authenticationType,
+    customerNumber,
+    phoneNumber: authPhoneNumber,
+  } = useAuthState();
+  const {
+    mutate: updateProfile,
+    isLoading: isLoadingUpdateProfile,
+    isError: isErrorUpdateProfile,
+    isSuccess: isSuccessUpdateProfile,
+    error: errorUpdate,
+  } = useProfileUpdateMutation();
+  const {
+    data: customerProfile,
+    isLoading: isLoadingGetProfile,
+    isError: isErrorGetProfile,
+    refetch: refetchProfile,
+    isRefetching: isRefetchingProfile,
+  } = useProfileQuery();
+
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [surname, setSurname] = useState('');
+  const [invalidEmail, setInvalidEmail] = useState<boolean>(false);
+  const isLoadingOrSubmittingProfile =
+    isLoadingUpdateProfile || isLoadingGetProfile;
+
+  const phoneNumber = parsePhoneNumber(
+    authPhoneNumber ?? '',
+  )?.formatInternational();
+
+  const onSubmit = async () => {
+    if (isValidEmail(email)) {
+      updateProfile({firstName, surname, email});
+    } else {
+      setInvalidEmail(true);
+    }
+  };
+
+  const getEmailErrorText = (
+    invalidEmail: boolean,
+    errorOnUpdate: any,
+  ): string | undefined => {
+    if (errorOnUpdate?.status === 602) {
+      return t(EditProfileTexts.personalDetails.email.unavailableError);
+    } else if (invalidEmail) {
+      return t(EditProfileTexts.personalDetails.email.formattingError);
+    }
+  };
+
+  useEffect(() => {
+    if (customerProfile) {
+      const profile: CustomerProfile = customerProfile;
+      const nonUnderscoreString = (str: string) => (str === '_' ? '' : str);
```

Comment:
Some doc here? As it is kinda hard to understand why it is being done when people doesn't know that this underscore stuff is a "hack" at Entur.